prow/github/client.go

```diff
@@ -883,12 +883,26 @@ func (c *client) requestRetry(method, path, accept, org string, body interface{}
 						resp.Body.Close()
 						break
 					}
-				} else if oauthScopes := resp.Header.Get("X-Accepted-OAuth-Scopes"); len(oauthScopes) > 0 {
+				} else {
+					acceptedOAuthScopes := sets.NewString(strings.Split(resp.Header.Get("X-Accepted-OAuth-Scopes"), ",")...)
 					authorizedScopes := resp.Header.Get("X-OAuth-Scopes")
 					if authorizedScopes == "" {
 						authorizedScopes = "no"
 					}
-					err = fmt.Errorf("the account is using %s oauth scopes, please make sure you are using at least one of the following oauth scopes: %s", authorizedScopes, oauthScopes)
+					oauthScopeAccepted := false
+					for _, scope := range strings.Split(authorizedScopes, ",") {
+						if acceptedOAuthScopes.Has(scope) {
+							oauthScopeAccepted = true
+							break
+						}
+					}
+
+					if !oauthScopeAccepted {
+						err = fmt.Errorf("the account is using %s oauth scopes, please make sure you are using at least one of the following oauth scopes: %s", authorizedScopes, acceptedOAuthScopes)
+					} else {
+						body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
It would be nice if there was a string in the error message we could key off, rather than needing to compare accepted & supplied scopes for that error.